chaincode/ledger_db.go

```diff
@@ -240,6 +289,81 @@ func (db *LedgerDB) GetTraintuple(key string) (Traintuple, error) {
 	return traintuple, nil
 }
 
+// GetCompositeTraintuple fetches a CompositeTraintuple from the ledger using its unique key
+func (db *LedgerDB) GetCompositeTraintuple(key string) (CompositeTraintuple, error) {
+	traintuple := CompositeTraintuple{}
+	if err := db.Get(key, &traintuple); err != nil {
+		return traintuple, err
+	}
+	if traintuple.AssetType != CompositeTraintupleType {
+		return traintuple, errors.NotFound("traintuple %s not found", key)
+	}
+	return traintuple, nil
+}
+
+// GetAggregatetuple fetches a Aggregatetuple from the ledger using its unique key
+func (db *LedgerDB) GetAggregatetuple(key string) (Aggregatetuple, error) {
+	aggregatetuple := Aggregatetuple{}
+	if err := db.Get(key, &aggregatetuple); err != nil {
+		return aggregatetuple, err
+	}
+	if aggregatetuple.AssetType != AggregatetupleType {
+		return aggregatetuple, errors.NotFound("aggregatetuple %s not found", key)
+	}
+	return aggregatetuple, nil
+}
+
+// CompositeModelType is a type of model generated by a composite traintuple
+type CompositeModelType uint8
+
+// Const representing the types models generated by a composite traintuple
+const (
+	HeadType CompositeModelType = iota
+	TrunkType
+)
+
+// GetOutModelHashDress retrieves an out-Model from a traintuple key.
+// Return an error if the traintupleKey was not found.
+func (db *LedgerDB) GetOutModelHashDress(traintupleKey string, modelType CompositeModelType, allowedAssetTypes []AssetType) (*HashDress, error) {
+	for _, assetType := range allowedAssetTypes {
+		switch AssetType(assetType) {
+		case CompositeTraintupleType:
+			tuple, err := db.GetCompositeTraintuple(traintupleKey)
+			if err == nil {
+				switch modelType {
+				case HeadType:
+					return tuple.OutHeadModel.OutModel, nil
+				case TrunkType:
+					return tuple.OutTrunkModel.OutModel, nil
+				default:
+					return nil, fmt.Errorf("GetOutModelHashDress: Unsupported composite model type %s", modelType)
+				}
+			}
+		case TraintupleType:
+			tuple, err := db.GetTraintuple(traintupleKey)
+			if err == nil {
+				return tuple.OutModel, nil
+			}
+		case AggregatetupleType:
+			tuple, err := db.GetAggregatetuple(traintupleKey)
+			if err == nil {
+				return tuple.OutModel, nil
+			}
+		default:
+			return nil, fmt.Errorf("GetOutModelHashDress: Unsupported asset type %s", assetType)
+		}
+	}
+
+	allowedStr := ""
+	for idx, assetType := range allowedAssetTypes {
+		if idx != 0 {
+			allowedStr = allowedStr + ", "
+		}
+		allowedStr = allowedStr + assetType.String()
+	}
+	return nil, errors.NotFound("GetOutModelHashDress: Could not find traintuple %s with key \"%s\". Allowed types: {%s}.", modelType, traintupleKey, allowedStr)
```

Comment:
Maybe use `%v` format verb so you don't have to format `allowedStr` by hand:
```suggestion
	return nil, errors.NotFound(
	"GetOutModelHashDress: Could not find traintuple %s with key \"%s\". Allowed types: %v.",
	modelType,
	traintupleKey,
	allowedAssetTypes)
```